virttest/qemu_vm.py

```diff
@@ -896,25 +896,35 @@ def add_memorys(devices, params):
 
             :param devices: VM devices container
             """
-            options = []
+            options, devs = [], []
+            params.setdefault("mem", "512")
             mem_params = params.object_params("mem")
-            if mem_params.get("maxmem"):
+            mem_params.setdefault("automem", "yes")
+            automem = mem_params["automem"] == "yes"
+            if automem:
                 normalize_data_size = utils_misc.normalize_data_size
-                values = map(lambda x: "%sM" % mem_params.get(x, 512), ["maxmem", "mem"])
-                values = map(lambda x: int(float(normalize_data_size(x))), values)
-                params["mem"] = min(values)
-            options.append("%s" % params.get("mem", 512))
-            if mem_params.get("slots") and mem_params.get("maxmem"):
-                options.append("slots=%s" % mem_params["slots"])
-                options.append("maxmem=%s" % mem_params["maxmem"])
-
-            cmdline = "-m %s" % ",".join(options)
-            dev = StrDev("mem", cmdline=cmdline)
-            devices.insert(dev)
-            for name in mem_params.objects("mem_devs"):
-                memdev_params = mem_params.object_params(name)
-                mem_devs = devices.memory_define_by_params(memdev_params, name)
-                devices.insert(mem_devs)
+                mem_keys = ["maxmem", "mem", "max_usable_mem", "maxmem_mem"]
+                mem_values = [mem_params[x] for x in mem_keys if mem_params.get(x)]
+                mem_values = map(
+                    lambda x: int(
+                        float(
+                            normalize_data_size(
+                                '%sM' %
+                                x))),
+                    mem_values)
+                params["mem"] = min(mem_values)
```

Comment:
And why does the memory size has to be set as the minimum value of the above params?